src/python/pants/backend/python/tasks/resolve_requirements.py

```diff
@@ -19,6 +19,7 @@ def product_types(cls):
 
   def execute(self):
     req_libs = self.context.targets(has_python_requirements)
-    if req_libs:
+    dist_tgts = self.context.targets(is_local_python_dist)
+    if req_libs or dist_tgts:
```

Comment:
I'm not seeing how `dist_tgts` is used here?

..but maybe `has_python_requirements` should be expanded to also match `PythonDistribution` targets if they now provide a resolvable requirement on the surface?

Reply:
This would not be viable in the current implementation fwict, as `PythonDistribution` targets would get treated as requirements libraries elsewhere in the codebase and break things. I see the need for the distinction because tasks that operate on req libs need `PythonRequirementsLibrary` targets, and while these are provided by py dist targets, it appears to me that the current backend does not accept `PythonDistribution` targets as a `has_python_requirements` target as of right now.